astropy/io/fits/tests/test_core.py

```diff
@@ -1015,6 +1017,34 @@ def flush(self):
             mmap.mmap = old_mmap
             _File.__dict__['_mmap_available']._cache.clear()
 
+    @pytest.mark.openfiles_ignore
+    def test_mmap_allocate_error(self):
+        """
+        Regression test for https://github.com/astropy/astropy/issues/1380
+
+        Temporarily patches mmap.mmap to raise an OSError if mode is ACCESS_COPY.
+        """
+
+        mmap_original = mmap.mmap
+
+        # We patch mmap here to raise an error if access=mmap.ACCESS_COPY, which
+        # emulates an issue that an OSError is raised if the available address
+        # space is less than the size of the file even if memory mapping is used.
+
+        def mmap_patched(*args, **kwargs):
+            if kwargs.get('access') == mmap.ACCESS_COPY:
+                exc = OSError()
+                exc.errno = errno.ENOMEM
+                raise exc
+            else:
+                return mmap_original(*args, **kwargs)
+
+        with fits.open(self.data('test0.fits'), memmap=True) as hdulist:
+            with patch.object(mmap, 'mmap', side_effect=mmap_patched) as p:
```

Comment:
Might be completely off-base here, and I'm not sure whether this will fix the open files issue or not, but it seems like the order of these context handlers might need to be reversed?

Reply:
We've already tried every permutation, and it doesn't help

Reply:
After looking at this myself for a little while, I think I'm willing to say that this is probably just due to some weird interaction between `mock` and `mmap`, and so skipping the open files test here is okay.